### PR TITLE
Change from device.get_input_report() to device.get_feature_report()

### PR DESCRIPTION
**Pull request by panmarco83**
This is all that was needed for this to work (with sudo) on ubuntu 20.04.02.

### CODE REVIEW

File: src/main/python/intellimouse.py [hunk 1]
```diff
@@ -69,7 +69,14 @@ def __read_property(self, property):
 			raise TypeError("please make sure to pass a integer for the property argument...")
 		report = self.__pad_right([self.__WRITE_REPORT_ID, property], self.__WRITE_REPORT_LENGTH)
 		self.__device.send_feature_report(report)
-		result = self.__device.get_input_report(self.__READ_REPORT_ID, self.__READ_REPORT_LENGTH)
+
+		# I really don't know what OS uses .get_input_report() instead of .get_feature_report().
+		# Here on ubuntu 20.04.2, it's "_feature_" instead of "_input_"
```
Comment (K-Visscher) [inline src/main/python/intellimouse.py:74]: I'm not quite sure about this, I kinda vageuly remember the Microsoft Mouse and Keyboard Center explicitly sending an input report and not an feature report, but maybe there's different versions of the mouse' firmware around 🤔.

Is okay with you to wait a bit, until I've setup an Ubuntu VM for this to get tested?

**Comment by panmarco83** [comment]
I'm fine with waiting.
Good luck

**Comment by K-Visscher** [comment]
Hi @panmarco83,

I've set-up an VM with `Ubuntu 20.04.03` and connected my Pro IntelliMouse to it.
And then ran the ProIntellimouse tests from the [classic-intellimouse-support branch](https://github.com/K-Visscher/pro-intellimouse-control-panel/tree/classic-intellimouse-support).
i.e: 
```bash
sudo pip install .
sudo python3 -m unittest -v tests.test_pro_intellimouse.TestProIntelliMouse
```
The test output was the following:

test_output
In other words, I'm unable to reproduce your problem.

I think you might be using a different backend (e.g. hidraw), in general use different dependencies or have different firmware version on the Pro IntelliMouse. 
Could you run the following commands and post the output?
backend_version

```bash
sudo apt list --installed | grep libusb
sudo apt list --installed | grep hidapi
sudo apt list --installed | grep hidraw
```

Could if you by any chance have Windows machine lying around? also share the following screenshot:
firmware_intellimouse

**Comment by panmarco83** [comment]
**sudo apt list --installed | grep libusb**

libhidapi-libusb0/focal,now 0.10.1+dfsg-1\~20.04.sav0 amd64 [installed,automatic]
libusb-1.0-0-dev/focal,now 2:1.0.23-2build1 amd64 [installed]
libusb-1.0-0/focal,now 2:1.0.23-2build1 amd64 [installed,automatic]
libusb-1.0-0/focal,now 2:1.0.23-2build1 i386 [installed,automatic]
libusb-1.0-doc/focal,focal,now 2:1.0.23-2build1 all [installed,automatic]
libusbmuxd6/focal,now 2.0.1-2 amd64 [installed,automatic]
libusbredirhost1/focal,now 0.8.0-1 amd64 [installed,automatic]
libusbredirparser1/focal,now 0.8.0-1 amd64 [installed,automatic]
python3-libusb1/focal,focal,now 1.7-2 all [installed]


**sudo apt list --installed | grep hidapi**

libhidapi-dev/focal,now 0.10.1+dfsg-1\~20.04.sav0 amd64 [installed]
libhidapi-hidraw0/focal,now 0.10.1+dfsg-1\~20.04.sav0 amd64 [installed,automatic]
libhidapi-libusb0/focal,now 0.10.1+dfsg-1\~20.04.sav0 amd64 [installed,automatic]
python3-hidapi/focal,now 0.2.2-1.1build1 amd64 [installed]



**sudo apt list --installed | grep hidraw**

libhidapi-hidraw0/focal,now 0.10.1+dfsg-1~20.04.sav0 amd64 [installed,automatic]

**Comment by K-Visscher** [comment]
I installed the same dependencies as you and then I installed hidapi and ran my tool, I still couldn't reproduce it. 
Did you build the python hidapi binding from source and pass any parameters? (my guess might be that you're using hidraw as a backend instead of libusb).

Also what happens, do you get an error message, if so could you share it? 
Do you have steps to reproduce?

**Comment by panmarco83** [comment]
First, have you fully updated the VM ? sudo apt update; sudo apt upgrade; sudo apt dist-upgrade

**Comment by K-Visscher** [comment]
Yes, the VM is fully updated.

> All I did was clone the repo then cd into the python folder src/main/python/ then start the script with sudo python3 main.py

Could you maybe please try again with the classic-intellimouse-support branch?
and if you encounter an error post it?

usage example:
```bash
intellimouse-ctl get --dpi 0
```
I'm also curious what happens if you uninstall hidraw, and just only have libusb?

**Pull request closed without merging**